src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasService.kt

```diff
@@ -22,47 +22,41 @@ import io.grpc.stub.StreamObserver
  */
 class CosmasService : CosmasGrpc.CosmasImplBase() {
 
-    private val files: HashMap<String, HashMap<String, ArrayList<ByteString>>> = HashMap()
-    
+    private val files = mutableMapOf<String, MutableList<ByteString>>()
+
     override fun getVersion(request: CosmasProto.GetVersionRequest,
                             responseObserver: StreamObserver<CosmasProto.GetVersionResponse>) {
         println("Get request for version: ${request.version}")
-        val response: CosmasProto.GetVersionResponse
-        response = if (!checkRequest(request)) {
-            println("Bad request")
-            CosmasProto.GetVersionResponse.newBuilder().build()
+        val response = CosmasProto.GetVersionResponse.newBuilder()
+        val fileVersions = files[request.fileId]
+        if (fileVersions != null &&
+                request.version < fileVersions.size &&
+                request.version >= 0) {
+            val file = fileVersions[request.version]

```

Comment:
you don't use this `file` variable except for immediate assignment to other variable in the next line. Just remove it.

Reply:
done